mockrxandroidble/src/main/java/com/polidea/rxandroidble/mockrxandroidble/RxBleClientMock.java

```diff
@@ -271,18 +277,25 @@ public RxBleDevice getBleDevice(@NonNull String macAddress) {
         return createScanOperation(filterServiceUUIDs);
     }
 
+    public void simulateDeviceDiscovery(@NonNull RxBleDeviceMock deviceMock) {
```

Comment:
Another maybe even nicer option would be to add a constructor that directly takes an `Observable` that gets assigned to the `discoveredDevicesSubject` field. That would give users complete freedom on when which devices appear and disappear (and maybe even the scan error stuff?).
I just tried to stick close to the current implementation in order to increase the chances to get this in ;)